src/Quantum_Computations_Utilities.py

```diff
@@ -54,9 +56,13 @@ def count_tv_distance_error_bound_of_experiment_results(outcomes_number: int, sa
         :return: Bound on the tv distance between the estimate and the experimental results.
     """
     possibly_huge_number = 2 ** outcomes_number - 2
-    prime_factors_of_the_number = get_prime_factors(possibly_huge_number)
+    if not isinf(possibly_huge_number):
+        prime_factors_of_the_number = get_prime_factors(possibly_huge_number)
+    else:
+        # In case the number is too big for python to process we need to approximate
```

Comment:
`possibly_huge_number` has to be an integer for me to be able to get its prime factors, so yes I expect it to be one.

The whole point behind this `math.isinf` stuff was that during the tests I've occurred the infinity once. Maybe there was some other error that caused it and the infinity can not in fact occur here?

As a matter of fact, I've just tested what you've suggested. It's far larger than the number of the outcomes we expect during the experiments (I mean d, obviously) and returned memory error. For d = 2 ** 16 was big and it actually didn't return inf but a huge integer. I guess something just went wrong elsewhere. I dunno.

Given that this method is for classical simulation and has to be as fast as I can get it I rewrote this method to compute approximate bound. It's now way more readable. I'm hinting at the approximation in the docstring and the in-code comment.

Reply:
If you got infinity once, it means that some part of your code passed a floating point down the road. Any multiplication by a float will propagate floats. It could be worth leaving a breakpoint and finding out from where did the float come. You could also leave a type check, whenever the value is not an int in pythonic way, print stacktrace, so that you can determine where did this value come from.